### PR TITLE
Iteration 3 methods

### PR DESCRIPTION
**Pull request by whitneykidd**
Created methods and helper methods:
games_by_team
unique_team_ids
total_games_by_team_id
average_goals_per_team
best_offense
worst_offense

Created and passed tests:
test_games_by_team
test_total_games_by_team_id
test_total_goals_by_team_id
test_returns_average_goals
test_best_offense
test_worst_offense

### CODE REVIEW

File: lib/game_teams_stats.rb [hunk 1]
```diff
@@ -7,6 +8,38 @@ class GameTeamStats
...
+  def total_goals_by_team_id(team_id)
+      games_by_team(team_id).sum {|game_team| game_team.goals}
+  end
+
+  def average_goals_per_team(team_id)
+    total_goals_by_team_id(team_id) / total_games_by_team_id(team_id)
+  end
+
+  def best_offense
+    team_id = unique_team_ids.max_by { |team_id| average_goals_per_team(team_id) }
```
Comment (Maxwell-Baird) [inline lib/game_teams_stats.rb:35]: I like how you built all these helper methods.

Comment (tylertomlinson) [reply lib/game_teams_stats.rb]: She's heating up!

File: lib/game_teams_stats.rb [hunk 2]
```diff
@@ -7,6 +8,38 @@ class GameTeamStats
...
+    games_by_team(team_id).length
+  end
+
+  def total_goals_by_team_id(team_id)
+      games_by_team(team_id).sum {|game_team| game_team.goals}
+  end
+
+  def average_goals_per_team(team_id)
+    total_goals_by_team_id(team_id) / total_games_by_team_id(team_id)
+  end
```
Comment (tylertomlinson) [inline lib/game_teams_stats.rb]: She's heating up!

File: lib/game_teams_stats.rb [hunk 3]
```diff
@@ -7,6 +8,38 @@ class GameTeamStats
...
+
+  def best_offense
+    team_id = unique_team_ids.max_by { |team_id| average_goals_per_team(team_id) }
+    @team_stats.find_name(team_id)
+  end
+
+  def worst_offense
+    team_id = unique_team_ids.min_by { |team_id| average_goals_per_team(team_id) }
+    @team_stats.find_name(team_id)
+  end
```
Comment (tylertomlinson) [inline lib/game_teams_stats.rb:42]: She's on FIRE!

File: test/game_teams_stats_test.rb [hunk 4]
```diff
@@ -25,4 +25,32 @@ def test_attributes_for_instance_of_game_teams_within_game_team_stats
...
+    assert_equal 2, @game_team_stats.average_goals_per_team(1)
+  end
+
+  def test_best_offense
+    assert_equal "FC Dallas", @game_team_stats.best_offense
+  end
+
+  def test_worst_offense
+    assert_equal "FC Cincinnati", @game_team_stats.worst_offense
+  end
```
Comment (tylertomlinson) [inline test/game_teams_stats_test.rb]: Solid tests. Might want to add more to our fixtures so we can use different Ids with a wider variety of games. Something to note

Comment (whitneykidd) [reply test/game_teams_stats_test.rb]: Totally agree

**Pull request merged**